apps/web/src/hooks/api/posts/useAdmireToken.ts

```diff
@@ -0,0 +1,142 @@
+import { useCallback } from 'react';
+import { ConnectionHandler, graphql } from 'react-relay';
+import { SelectorStoreUpdater } from 'relay-runtime';
+
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { useAdmireTokenMutation } from '~/generated/useAdmireTokenMutation.graphql';
+import { AdditionalContext, useReportError } from '~/shared/contexts/ErrorReportingContext';
+import { usePromisifiedMutation } from '~/shared/relay/usePromisifiedMutation';
+import { OptimisticUserInfo } from '~/utils/useOptimisticUserInfo';
+
+export default function useAdmireToken() {
```

Comment:
question: noticed on web we have `useAdmireToken` and `useRemoveTokenAdmire`. on mobile we have `useToggleTokenAdmire`, both with a lot of shared logic

does it make sense to consolidate everything under the same hook in `/shared`?